test/tests.js

```diff
@@ -107,6 +107,32 @@ describe('function tests', function () {
     expect(arg(f)).to.deep.equal(['a']);
   })
 
+  it('supports ES2015 fat arrow function without parens test1.', function() {
+    var f = c => {
+      var test2 = c.resolve();
+      return new Test3(test2);
+    };
+
+    expect(arg(f)).to.deep.equal(['c']);
+  })
+
+  it('supports ES2015 fat arrow function without parens test2.', function() {
+    var f = a => {
+      return new Promise((resolve, reject) => {
+        setTimeout(() => resolve(a * 2), 500);
+      });
+    }
+
+    expect(arg(f)).to.deep.equal(['a']);
+  });
+
+  it('supports ES2015 fat arrow function without parens test3.', function() {
+    var f = items => items.map(
```

Comment:
Wrap fat arrow functions as strings to run on legacy node systems.  See other tests for example